burn-tensor/src/tensor/api/base.rs

```diff
@@ -530,6 +530,184 @@ where
+    /// * `pad_mode` - The padding mode, determining how the padding is applied. Currently, only `PadMode::Constant` is supported.
+    /// * `value` - The value to be used for padding in constant mode.
+    ///
+    /// # Returns
+    ///
+    /// A new `Tensor<B, D, K>` instance with the specified padding applied.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Data,PadMode,Padding,Shape,Tensor};
+    ///
+    /// fn pad_example<B: Backend>()
+    /// where
+    ///     B::FloatElem: From<f32>, // Ensure that FloatElem can be created from a floating-point number
+    /// {
+    ///    let pad_const = 1.1;
+    ///    let tensor = Tensor::<B, 4>::ones_devauto(Shape::new([2, 3, 4, 5]));
+    ///    let padding = Padding::uniform(2);
+    ///
+    ///    let padded_tensor = tensor.pad(padding, PadMode::Constant, Some(pad_const.into()));
+    /// }    
+    ///
+    ///
+    /// ```
+    /// Pads the tensor with the specified padding configuration and padding mode.
+    pub fn pad(
+        self,
+        pad_amount: Padding,
+        pad_mode: PadMode,
+        value: Option<K::Elem>,
+    ) -> Tensor<B, D, K>
+    where
+        K::Elem: Element,
+    {
+        match pad_mode {
+            PadMode::Constant => {
+                let padding_value = value.expect("Padding value required for Constant mode");
+                let mut padded_dims: [usize; D] = self.dims();
+
+                // Update the last two dimensions with padding
+                padded_dims[D - 2] += pad_amount.top + pad_amount.bottom;
+                padded_dims[D - 1] += pad_amount.left + pad_amount.right;
+
+                // Initialize ranges for all dimensions
+                let ranges: [Range<usize>; D] = padded_dims
+                    .iter()
+                    .enumerate()
+                    .map(|(i, &dim)| {
+                        if i == D - 2 {
+                            // Second last dimension
+                            pad_amount.top..dim - pad_amount.bottom
+                        } else if i == D - 1 {
+                            // Last dimension
+                            pad_amount.left..dim - pad_amount.right
+                        } else {
+                            0..dim // Other dimensions remain unchanged
+                        }
+                    })
+                    .collect::<Vec<Range<usize>>>()
+                    .try_into()
+                    .unwrap();
+
+                // Create the padded tensor
+                let padded_shape = Shape::from(padded_dims);
+                let padded_data = Data::full(padded_shape, padding_value);
+                let padded_tensor = Tensor::<B, D, K>::from_data_devauto(padded_data);
+
+                // Assign the original tensor data to the appropriate slice of the padded tensor
+                padded_tensor.slice_assign(ranges, self)
```

Comment:
Using `Data` is not efficient; it's preferable to use `Tensor::full(shape, value)` instead of `Data::full`.